test/timeout.js

```diff
@@ -458,3 +458,17 @@ test('no more timeouts after an error', async t => {
 	// Wait a bit more to check if there are any unhandled errors
 	await delay(10);
 });
+
+test('socket timeout is canceled on error', async t => {
+	const message = 'oh, snap!';
+
+	const promise = got(s.url, {
+		timeout: {socket: requestTimeout},
+		retry: 0
+	}).on('request', request => {
+		request.emit('error', new Error(message));
+	});
+
+	await t.throwsAsync(promise, {message});
+	await delay(10);
```

Comment:
Consider copying the note here as well. Wasn't immediately clear why we were waiting. 

> // Wait a bit more to check if there are any unhandled errors